packages/projection/src/operators/Mappers/withHandles.ts

```diff
@@ -25,21 +28,43 @@ export interface WithHandles {
 const handleFromAssetId = (assetId: Cardano.AssetId) =>
   Buffer.from(Cardano.AssetId.getAssetName(assetId), 'hex').toString('utf8');
 
+const logger: Logger = dummyLogger;
+
+const isValidHandle = (handle: string) => {
+  const pattern = /^[\w.-]+$/;
+  return pattern.test(handle);
+};
+
+const mapHandleToData = (
+  assetId: Cardano.AssetId,
+  data: { datum?: HexBlob | undefined; address?: Cardano.PaymentAddress; policyId: Cardano.PolicyId }
+) => {
+  const handle = handleFromAssetId(assetId);
+  if (!isValidHandle(handle)) throw new HandleParsingError(handle);
+  const { datum, address, policyId } = data;
+
+  return {
+    assetId,
+    datum,
+    handle,
+    latestOwnerAddress: address || null,
+    policyId
+  };
+};
+
 const getOutputHandles = (outputs: Cardano.TxOut[], policyIds: FilterByPolicyIds['policyIds']) => {
   const handles: Record<string, Handle> = {};
   for (const { address, value, datum } of outputs) {
     if (!value.assets) continue;
     for (const [assetId] of value.assets.entries()) {
       const policyId = Cardano.AssetId.getPolicyId(assetId);
       if (!policyIds.includes(policyId)) continue;
-      const handle = handleFromAssetId(assetId);
-      handles[handle] = {
-        assetId,
-        datum,
-        handle,
-        latestOwnerAddress: address,
-        policyId
-      };
+      try {
+        const handleData = mapHandleToData(assetId, { address, datum, policyId });
+        handles[handleData.handle] = handleData;
+      } catch (error: any) {
+        logger.error(error.message);
```

Comment:
This is a good practice because technically you can throw anything in js. Same in `getBurnedHandles` - probably a good idea to create a new `logError` util function.
```suggestion
      } catch (error: unknown) {
        const message = isError(error) ? error.message : 'Internal error: ' + error;
        logger.error(message);
```

Might no longer need a file-level `/* eslint-disable @typescript-eslint/no-explicit-any */` too then

Reply:
Fixed on 6c65203